### PR TITLE
save temporary task a user is typing

### PR DESCRIPTION
**Pull request by bryvin**
https://github.com/standardnotes/bounties/issues/8

Saves the text input of the "unsaved" task that the user has not hit 'enter' on yet to put into Open tasks

### CODE REVIEW

**Comment by mobitar** [comment]
Nice work, will give this a run sometime this week.

**Comment by bryvin** [comment]
You are correct, I just committed a fix for this. It was happening because I was setting the state in the actual component and also up in the parent. This lead to conflicting state depending on timing I guess.

I think this also fixed an issue where you could not backspace while creating a new task.

File: app/components/CreateTask.js [hunk 1]
```diff
@@ -1,32 +1,43 @@
...
+      var rawString = event.target.value;
+      this.submitTask(rawString);
     }
   }
 
-  submitTask() {
-    this.props.onSubmit(this.state.rawString);
-    this.setState({rawString: ''})
+  submitTask(value) {
+    TasksManager.get().setUnsavedTask('');
```
Comment (mobitar) [inline app/components/CreateTask.js]: Why not call this in Tasks.js in `createTask`? Would remove the TasksManager dependency from this class.

Comment (mobitar) [reply app/components/CreateTask.js]: No worries, I just did to my fork of your PR. Just wanted to make sure I wasn't overlooking something obvious.

**Comment by mobitar** [comment]
Merged, nice work. Btw, check out this bug I found with presave. Pretty cool bug :) https://github.com/sn-extensions/simple-task-editor/commit/cbf31721781ce07fe12604d28211559d70bb8984

**Comment by bryvin** [comment]
Oh interesting.. that does make sense. I'll keep that in mind, but how did you find that out lol

**Comment by mobitar** [comment]
User found it. If you type "12345" in a consistent rhythm with about 250ms in between each keystroke in any editor with presave, the "5" will not save. This was the cause.

**Pull request merged**